src/mongocrypt-marking.c

```diff
@@ -457,13 +457,39 @@ _mongocrypt_fle2_placeholder_to_insert_update_ciphertext (
 
    _mongocrypt_buffer_from_iter (&value, &placeholder->v_iter);
 
-   if (!_mongocrypt_fle2_placeholder_common (kb,
-                                             &common,
-                                             &placeholder->index_key_id,
-                                             &value,
-                                             true, /* derive tokens using counter */
-                                             placeholder->maxContentionCounter,
-                                             status)) {
+   int64_t contentionFactor = 0;
+   if (placeholder->maxContentionCounter < 0) {
```

Comment:
Shouldn't this check occur in the palceholder parse function?

Reply:
[_fle2_finalize_explicit](https://github.com/mongodb/libmongocrypt/blob/master/src/mongocrypt-ctx-encrypt.c#L1280) constructs a `_mongocrypt_marking_t` without parsing. This is a common code path between explicit and automatic encryption.

Moved this check to a common helper to error earlier.